src/test/librbd/mock/MockWatcher.h

```diff
@@ -0,0 +1,33 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_TEST_LIBRBD_MOCK_WATCHER_H
+#define CEPH_TEST_LIBRBD_MOCK_WATCHER_H
+
+#include "librbd/Watcher.h"
+#include "gmock/gmock.h"
+
+class Context;
+
+namespace librbd {
+
+struct MockWatcher : public Watcher {
```

Comment:
I'd prefer to keep the mocks as a clean, independent implementation (e.g. it doesn't half bootstrap a base class and override virtual methods). If you look at "journal/TypeTraits.h", you can see an example of how to derive the correct ```Watcher``` class based upon the type of ```ImageCtxT``` template parameter. You could add a ```librbd::watcher::Traits``` class in watcher/Types.h to accomplish the same thing.